tests/aws/services/cloudformation/v2/test_change_set_ref.py

```diff
@@ -309,3 +309,37 @@ def test_immutable_property_update_causes_resource_replacement(
             }
         }
         capture_update_process(snapshot, template_1, template_2)
+
+    @markers.aws.validated
+    def test_supported_pseudo_parameter(
+        self,
+        snapshot,
+        capture_update_process,
+    ):
+        topic_name = f"topic-name-1-{long_uid()}"
+        snapshot.add_transformer(RegexTransformer(topic_name, "topic-name"))
+        test_name = f"test-name-{long_uid()}"
+        snapshot.add_transformer(RegexTransformer(test_name, "test-name"))
+        template_1 = {
+            "Resources": {
+                "Topic": {"Type": "AWS::SNS::Topic", "Properties": {"TopicName": topic_name}},
+            }
+        }
```

Comment:
Is there a reason we create an sns topic in template one and use a completely different resource in template two?

Reply:
No, there is no particular reason other than needing an update with a resource listing a number of supported pseudo parameters. I changed it to only use SNS topics.